addons/web/static/src/js/views/list/list_confirm_dialog.js

```diff
@@ -68,25 +68,53 @@ const ListConfirmDialog = Dialog.extend(WidgetAdapterMixin, {
                 options: widgetOptions,
             });
         }
+
+        this.relatedFieldWidget = undefined;
+        if (this.fieldWidget.attrs.widget === 'daterange' && changes.relatedFieldName) {
+            const fieldName = changes.relatedFieldName;
+            const fieldLabel = record.fields[fieldName].string;
+            this.$content.find('table.o_modal_changes tbody').append(qweb.render('ListView.confirmModal.relatedField', { fieldLabel }));
+            const RelatedWidget = record.fieldsInfo.list[fieldName].Widget;
+
+            if (this.isLegacyWidget) { // the related widget is the same (= DateRange widget) then we don't need to recheck it
+                this.relatedFieldWidget = new RelatedWidget(this, fieldName, record, widgetOptions);
+            } else {
+                this.relatedFieldWidget = new FieldWrapper(this, Widget, {
+                    fieldName,
+                    record,
+                    options: widgetOptions,
+                });
+            }
+        }
     },
     /**
      * @override
      */
     willStart: function () {
-        let widgetProm;
+        const widgetPromises = [];
         if (this.isLegacyWidget) {
-            widgetProm = this.fieldWidget._widgetRenderAndInsert(function () {});
+            widgetPromises.push(this.fieldWidget._widgetRenderAndInsert(function () {}));
+            if (this.relatedFieldWidget) {
+                widgetPromises.push(this.relatedFieldWidget._widgetRenderAndInsert(function () {}));
+            }
         } else {
-            widgetProm = this.fieldWidget.mount(document.createDocumentFragment());
+            widgetPromises.push(this.fieldWidget.mount(document.createDocumentFragment()));
+            if (this.relatedFieldWidget) {
+                widgetPromises.push(this.relatedFieldWidget.mount(document.createDocumentFragment()));
+            }
         }
-        return Promise.all([widgetProm, this._super.apply(this, arguments)]);
+        return Promise.all([...widgetPromises, this._super.apply(this, arguments)]);
     },
     /**
      * @override
      */
     start: function () {
         this.$content.find('.o_changes_widget').replaceWith(this.fieldWidget.$el);
         this.fieldWidget.el.style.pointerEvents = 'none';
+        if (this.relatedFieldWidget) {
+            this.$content.find('.o_changes_related_widget').replaceWith(this.relatedFieldWidget.$el);
+            this.relatedFieldWidget.el.style.pointerEvents = 'none';
+        }
```

Comment:
It looks to me that we always duplicate the code logic from the fieldWidget to the relatefieldWidget, but on the other side does it really worth extracting the logic and calling it with the proper arg? I am not sure neither... Did you have the same feeling?

Reply:
Yeah, indeed !
If I can refactor it and thus to avoid the duplicate code, then I will do something like this:
```js
    willStart: function () {
        const widgetPromises = [this.fieldWidget, this.relatedFieldWidget].map((fieldWidget) => {
            if (!fieldWidget) return Promise.resolve();

            if (this.isLegacyWidget) {
                return fieldWidget._widgetRenderAndInsert(function () {});
            }
            return fieldWidget.mount(document.createDocumentFragment());
        });
        return Promise.all([...widgetPromises, this._super.apply(this, arguments)]);
    },
    /**
     * @override
     */
    start: function () {
        const replaceContentByFieldWidget = (elementSelector, fieldWidget) => {
            this.$content.find(elementSelector).replaceWith(fieldWidget.$el);
            fieldWidget.el.style.pointerEvents = 'none';
        };
        replaceContentByFieldWidget('.o_changes_widget', this.fieldWidget);
        if (this.relatedFieldWidget) {
            replaceContentByFieldWidget('.o_changes_related_widget', this.relatedFieldWidget);
        }
        return this._super.apply(this, arguments);
    },
```